au4/src/effects/effects_base/internal/effectsuiactions.cpp

```diff
@@ -31,6 +35,21 @@ EffectsUiActions::EffectsUiActions(std::shared_ptr<EffectsActionsController> con
 {
 }
 
+void EffectsUiActions::init()
+{
+    effectExecutionScenario()->lastProcessorIdChanged().onReceive(this, [this](const EffectId& effectId) {
+        const auto it = std::find_if(m_actions.begin(), m_actions.end(), [](const UiAction& action) {
+            return action.code == "repeat-last-effect";
+        });
+        IF_ASSERT_FAILED(it != m_actions.end()) {
+            return;
+        }
+        const auto effectTitle = effectsProvider()->meta(effectId).title;
```

Comment:
Can the last processor ID become invalid if the last processor executed was not the processor? Or is the processor ID always valid here, and it tells which processor executed last, not which effect last?